### PR TITLE
Fix #203 and #204: bend and release on same note

### LINKED ISSUE #203: Document whether a <note> can contain multiple <bend>s
I am investigating the MusicXML generated by [Guitar Pro 7](https://www.guitar-pro.com/) (released a few days ago), and I've found that it sometimes puts two `<bend>` elements within a single `<note>`.

This happens in case of a "bend and release" bend. For example, in this screenshot from a Guitar Pro score, the measure contains two notes, each of which has a "bend and release."

screen shot 2017-04-21 at 11 36 36 am

Here is Guitar Pro 7's resulting MusicXML (with irrelevant stuff stripped out):

```
 <note>
  <pitch>
   <step>E</step>
   <octave>4</octave>
  </pitch>
  <type>half</type>
  <notations>
   <technical>
    <bend>
     <bend-alter>2</bend-alter>
    </bend>
    <bend>
     <bend-alter>0</bend-alter>
     <release/>
    </bend>
    <string>2</string>
    <fret>5</fret>
   </technical>
  </notations>
 </note>
 <note>
  <pitch>
   <step>E</step>
   <octave>4</octave>
  </pitch>
  <type>half</type>
  <notations>
   <technical>
    <bend>
     <bend-alter>1</bend-alter>
    </bend>
    <bend>
     <bend-alter>0</bend-alter>
     <release/>
    </bend>
    <string>2</string>
    <fret>5</fret>
   </technical>
  </notations>
 </note>
```

My question is: if indeed it's correct to use multiple `<bend>` elements within a single note, can we improve the documentation to clarify this? I'm happy to tackle the documentation; just need to know whether it's correct in the first place. :-)

### LINKED ISSUE #204: Clarify whether <bend-alter> value is absolute or relative
I'm seeing conflicting real-world data for `<bend-alter>`. We could prevent inconsistency here by clarifying the documentation.

Consider a bend that's rhythmically split over two half notes. The note is bent a half step, then released to the original pitch:

screen shot 2017-04-21 at 11 52 25 am

What should the `<bend-alter>` value be for the second note (when the bend is released)? The two, mutually exclusive, approaches I've seen are:

* `<bend-alter>-1</bend-alter>` — relative to the _previous_ note. This is implied in the [MusicXML documentation](http://usermanuals.musicxml.com/MusicXML/MusicXML.htm#EL-MusicXML-bend-alter.htm).
* `<bend-alter>0</bend-alter>` — relative to the _original_ (unbent) pitch. This is how Guitar Pro 7 does it. Here is its MusicXML for the above example:

```
   <note>
    <pitch>
     <step>E</step>
     <octave>4</octave>
    </pitch>
    <duration>2</duration>
    <tie type="start"/>
    <type>half</type>
    <stem>up</stem>
    <notations>
     <technical>
      <bend>
       <bend-alter>1</bend-alter>
      </bend>
      <string>2</string>
      <fret>5</fret>
     </technical>
     <tied type="start"/>
    </notations>
   </note>
   <note>
    <pitch>
     <step>E</step>
     <octave>4</octave>
    </pitch>
    <duration>2</duration>
    <tie type="stop"/>
    <type>half</type>
    <stem>up</stem>
    <notations>
     <technical>
      <bend>
       <bend-alter>0</bend-alter>
       <release/>
      </bend>
      <string>2</string>
      <fret>5</fret>
     </technical>
     <tied type="stop"/>
    </notations>
   </note>
```

I cannot think of any particular advantages of either approach — we just need to clarify which one is the standard. Otherwise, bend data can't be guaranteed to be interoperable across programs.

### PR DESCRIPTION
**Pull request by mdgood**
This is the alternate approach to #391. In this approach the bend and release are on the same note rather than putting the release on an intermediate note.

### CODE REVIEW

File: schema/musicxml.xsd [hunk 1]
```diff
@@ -5320,6 +5316,17 @@ If the parentheses attribute is set to yes, the notehead is parenthesized. It is
 		</xs:simpleContent>
 	</xs:complexType>
 
+	<xs:complexType name="release">
+		<xs:annotation>
+			<xs:documentation>The release type indicates that a bend is a release rather than a normal bend or pre-bend. The offset attribute specifies where the release starts in terms of divisions. The first-beat and last-beat attributes of the parent bend element are relative to the original note position, not this offset value.</xs:documentation>
```
Comment (adrianholovaty) [inline schema/musicxml.xsd]: Is `offset` relative to the onset of this `<note>`, or relative to the global position? I think this is worth clarifying.

File: schema/musicxml.xsd [hunk 2]
```diff
@@ -4630,25 +4630,21 @@ The repeater attribute has been deprecated in MusicXML 3.0. Formerly used for tr
 
 	<xs:complexType name="bend">
 		<xs:annotation>
-			<xs:documentation>The bend type is used in guitar notation and tablature. The bend-alter element indicates the number of steps in the bend, similar to the alter element. As with the alter element, numbers like 0.5 can be used to indicate microtones. Negative numbers indicate pre-bends or releases; the pre-bend and release elements are used to distinguish what is intended. The shape attribute distinguishes between the angled bend symbols commonly used in standard notation and the curved bend symbols commonly used in both tablature and standard notation.</xs:documentation>
+			<xs:documentation>The bend type is used in guitar notation and tablature. Notes with a bend and release will have two bend elements, with the release following the bend. The shape attribute distinguishes between the angled bend symbols commonly used in standard notation and the curved bend symbols commonly used in both tablature and standard notation.</xs:documentation>
```
Comment (adrianholovaty) [inline schema/musicxml.xsd]: > Notes with a bend and release will have two bend elements, with the release following the bend.

The term "bend" in this sentence feels a bit overloaded and might be confusing; it's being used both to mean the concept and the MusicXML element. What about rewording to something like this:

"If a single note contains a bend and release, use two bend elements, one after another -- one element to represent the bend and another element to represent the release."

**Comment by adrianholovaty** [comment]
I've added a few comments with suggested wording tweaks. Otherwise this looks good to me!

I also wanted to point out for the record that this approach (as opposed to #391) does not give people the ability to specify the note spelling of the intermediary bend note. But I think that's totally OK. If people want to do that, they should just use two `<note>` elements instead of shoving the bend and release into a single `<note>`.

**Comment by mdgood** [comment]
Thanks @adrianholovaty! We could probably fix the issue with the intermediary bend note by adding a `diatonic` attribute to the `<release>` element that specifies the number of diatonic steps for the release. That should allow for the spelling of the intermediary note without using a separate `<note>` element. Does that seem like a good idea?

Added after: 

Looking at some more repertoire examples, I don't think this is a good idea. It seems it would be better to use two `<note>` elements as you say. Let me see if there is any documentation that should perhaps be clarified for this.

**Comment by mdgood** [comment]
@adrianholovaty, I have updated the pull request with these wording changes. Could you please let me know if those look good? It turned out there were no additional changes needed regarding an intermediary bend note.

**Pull request merged**